packages/medusa/src/api/routes/admin/discounts/add-items-to-condition-batch.ts

```diff
@@ -0,0 +1,130 @@
+ *                   type: string
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       import { DiscountConditionOperator } from "@medusajs/medusa"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.discounts.addConditionItemBatch(discount_id, condition_id, {
+ *         items: [{ id: item_id }]
+ *       })
+ *       .then(({ discount }) => {
+ *         console.log(discount.id);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request POST 'https://medusa-url.com/admin/discounts/{id}/conditions/{condition_id}/batch' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json' \
+ *       --data-raw '{
+ *           "items": [{ "id": "item_id" }]
+ *       }'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Discount Condition
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             discount:
+ *               $ref: "#/components/schemas/discount"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const { discount_id, condition_id } = req.params
+  const validatedBody =
+    req.validatedBody as AdminPostDiscountsDiscountConditionsConditionBatchReq
+
+  const conditionService: DiscountConditionService = req.scope.resolve(
+    "discountConditionService"
+  )
+  const manager: EntityManager = req.scope.resolve("manager")
+
+  const condition = await conditionService.retrieve(condition_id, {
+    select: ["id", "type", "discount_rule_id"],
+  })
+
+  const updateObj: UpsertDiscountConditionInput = {
+    id: condition_id,
+    rule_id: condition.discount_rule_id,
+    [DiscountConditionMapTypeToProperty[condition.type]]: validatedBody.items,
+  }
+
+  await manager.transaction(async (transactionManager) => {
+    await conditionService
+      .withTransaction(transactionManager)
+      .upsertCondition(updateObj, false)
+  })
+
+  const discountService: DiscountService = req.scope.resolve("discountService")
+  const discount = await discountService.retrieve(
+    discount_id,
+    req.retrieveConfig
+  )
+
+  res.status(200).json({ discount })
+}
+
+export class AdminPostDiscountsDiscountConditionsConditionBatchReq {
+  @IsArray()
+  items: { id: string }[]
+}
```

Comment:
thought: I think `items` is too ambiguous (given the notion of LineItems). Suggest going with `resource_ids` instead. Wdyt?

Reply:
you are right, did not thought about it, resource_ids is find with me, but it is an array of object which could in the future contains more than the ids no? otherwise I am good with that

Reply:
True `resources` then :)